pridepy/files/files.py

```
@@ -760,3 +762,239 @@ def get_all_category_file_list(self, accession: str, category: str):
+        skip_if_downloaded_already: bool,
+        max_connection_retries: int = 3,
+        max_download_retries: int = 3,
+    ) -> None:
+        """
+        Download a list of FTP URLs using a single connection, with retries and progress bars.
+        """
+        if not os.path.isdir(output_folder):
+            os.makedirs(output_folder, exist_ok=True)
+
+        def connect_ftp(host: str):
+            ftp = FTP(host, timeout=30)
```

Ruff (0.14.0)

FTP-related functions are being called. FTP is considered insecure. Use SSH/SFTP/SCP or some other encrypted protocol.

(S321)